### PR TITLE
Add more information to the NetworkState object

### LINKED ISSUE #36: Listing add extra Loading state?
For now , if I want ton know whether is loading more, fetch from refresh or network status seems unreasonable

### PR DESCRIPTION
**Pull request by matir91**
Fixes #36 

### Additional info:
- The network state now includes: `page`, `pageSize`, `isFirstPage` and `isLastPage`.

### CODE REVIEW

File: app/src/main/java/com/xmartlabs/sample/ui/NetworkStateItemViewHolder.kt [hunk 1]
```diff
@@ -27,10 +26,11 @@ class NetworkStateItemViewHolder(view: View,
     }
   }
   fun bindTo(networkState: NetworkState?) {
-    progressBar.visibility = toVisbility(networkState?.status == Status.RUNNING)
-    retry.visibility = toVisbility(networkState?.status == Status.FAILED)
-    errorMsg.visibility = toVisbility(networkState?.throwable?.message != null)
-    errorMsg.text = networkState?.throwable?.message
+    progressBar.visibility = toVisbility(networkState is NetworkState.Loading)
```
Comment (FerrariBruno) [inline app/src/main/java/com/xmartlabs/sample/ui/NetworkStateItemViewHolder.kt]: should be `toVisibility`

File: fountain/src/main/java/com/xmartlabs/fountain/feature/cachednetwork/BoundaryCallback.kt [hunk 2]
```diff
@@ -97,31 +100,52 @@ internal class BoundaryCallback<NetworkValue, DataSourceValue>(
...
+            NetworkState.Error(exception, firstPage, pagedListConfig.initialLoadSizeHint, true, isLastPage)
+        )
       }
     }
     return resetNetworkState
   }
 
+
+  private fun createLoadingState(page: Int, nextPage: Int, pageSize: Int): NetworkState.Loading {
+    val isLastPage = !networkDataSourceAdapter.canFetch(page = nextPage, pageSize = pagedListConfig.pageSize)
```
Comment (FerrariBruno) [inline fountain/src/main/java/com/xmartlabs/fountain/feature/cachednetwork/BoundaryCallback.kt]: can't you make an extension for this so we avoid having to repeat this line?

File: fountain/src/main/java/com/xmartlabs/fountain/feature/network/NetworkPagedDataSource.kt [hunk 3]
```diff
@@ -101,38 +114,48 @@ internal class NetworkPagedDataSource<T>(
     } else {
       retry = null
       val nextPage = firstPage + params.requestedLoadSize / pagedListConfig.pageSize
-      callback.onResult(initData!!, -1, nextPage)
-      networkState.postValue(NetworkState.LOADED)
-      initialLoad.postValue(NetworkState.LOADED)
+      callback.onResult(initData!!.getElements(), -1, nextPage)
```
Comment (FerrariBruno) [inline fountain/src/main/java/com/xmartlabs/fountain/feature/network/NetworkPagedDataSource.kt]: :O 

NPE's everywhere!!

Comment (matir91) [reply fountain/src/main/java/com/xmartlabs/fountain/feature/network/NetworkPagedDataSource.kt]: it cannot be null!

**Pull request merged**